AssertionError

### Hail version:

### What you did:
```
mt = hl.read_matrix_table(...)
    mt = mt.select_rows(mt.locus, mt.alleles)
    fam_kt = hl.import_fam(fam_file)
    mt = mt.annotate_cols(fam=fam_kt[mt.s])

    # Unphased for now, since mendel_errors does not support phased alleles
    mt = unphase_mt(mt)
    ped = hl.Pedigree.read(fam_file)
    tdt_table = hl.tdt(mt, ped)
    _, _, per_sample, per_variant = hl.mendel_errors(mt, ped)
    family_stats = Struct(mendel=per_variant[(mt.locus, mt.alleles)],
                          tdt=tdt_table[(mt.locus, mt.alleles)],
                          ac_unrelated_qc=hl.agg.sum(hl.agg.filter(
                              True & hl.is_missing(mt.fam.pat_id),
                              mt.GT.num_alt_alleles())),
                          meta={'group': 'raw'})
    mt = mt.annotate_rows(family_stats=[family_stats])

    mt = filter_to_adj(mt)
    adj_tdt_table = hl.tdt(mt, ped)
    _, _, adj_per_sample, adj_per_variant = hl.mendel_errors(mt, ped)
    family_stats_adj = Struct(mendel=adj_per_variant[(mt.locus, mt.alleles)],
                              tdt=adj_tdt_table[(mt.locus, mt.alleles)],
                              ac_unrelated_qc=hl.agg.sum(hl.agg.filter(
                                  True & hl.is_missing(mt.fam.pat_id),
                                  mt.GT.num_alt_alleles())),
                              meta={'group': 'adj'})

    per_sample = per_sample.annotate(adj=adj_per_sample[per_sample.s])

    mt = mt.annotate_rows(family_stats=mt.family_stats.append(family_stats_adj))
mt.write()
```
### What went wrong (all error messages here, including the full java stack trace):
```
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:156)
	at is.hail.annotations.RegionValueBuilder.addRegionValue(RegionValueBuilder.scala:394)
	at is.hail.annotations.RegionValueBuilder.addField(RegionValueBuilder.scala:335)
	at is.hail.variant.MatrixTable$$anonfun$dropEntries$2.apply(MatrixTable.scala:1433)
	at is.hail.variant.MatrixTable$$anonfun$dropEntries$2.apply(MatrixTable.scala:1421)
	at is.hail.variant.MatrixTable$$anonfun$95$$anonfun$apply$60$$anonfun$apply$4.apply$mcV$sp(MatrixTable.scala:1723)
	at is.hail.expr.types.TStruct$$anonfun$unsafeInsert$1.apply(TStruct.scala:177)
	at is.hail.expr.types.TStruct$$anonfun$unsafeInsert$1.apply(TStruct.scala:177)
	at is.hail.expr.types.TStruct$$anonfun$unsafeInsert$2.apply(TStruct.scala:197)
	at is.hail.expr.types.TStruct$$anonfun$unsafeInsert$2.apply(TStruct.scala:186)
	at is.hail.variant.MatrixTable$$anonfun$95$$anonfun$apply$60.apply(MatrixTable.scala:1722)
	at is.hail.variant.MatrixTable$$anonfun$95$$anonfun$apply$60.apply(MatrixTable.scala:1718)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.next(OrderedRVD.scala:736)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.next(OrderedRVD.scala:730)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.next(OrderedRVD.scala:736)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.next(OrderedRVD.scala:730)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.next(OrderedRVD.scala:736)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.next(OrderedRVD.scala:730)
	at scala.collection.Iterator$$anon$13.hasNext(Iterator.scala:462)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.hasNext(OrderedRVD.scala:733)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.hasNext(OrderedRVD.scala:733)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$6$$anon$5.hasNext(OrderedRVD.scala:733)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at is.hail.rvd.OrderedRVD$$anonfun$11.apply(OrderedRVD.scala:491)
	at is.hail.rvd.OrderedRVD$$anonfun$11.apply(OrderedRVD.scala:490)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsWithIndex$1$$anonfun$apply$26.apply(RDD.scala:844)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsWithIndex$1$$anonfun$apply$26.apply(RDD.scala:844)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:108)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:335)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/python/hail/tests/test_api.py
+++ b/python/hail/tests/test_api.py
@@ -529,12 +529,14 @@ class MatrixTests(unittest.TestCase):
         self.assertTrue('info' not in vds1.row)
         self.assertTrue('bar' not in vds1.col)
         self.assertTrue('GT' not in vds1.entry)
+        vds1._force_count_rows()
 
         vds2 = vds.drop(vds.GT, vds.info, vds.foo, vds.bar)
         self.assertTrue('foo' not in vds2.globals)
         self.assertTrue('info' not in vds2.row)
         self.assertTrue('bar' not in vds2.col)
         self.assertTrue('GT' not in vds2.entry)
+        vds2._force_count_rows()
 
     def test_drop_rows(self):
         vds = self.get_vds()

--- a/src/main/scala/is/hail/variant/MatrixTable.scala
+++ b/src/main/scala/is/hail/variant/MatrixTable.scala
@@ -1428,7 +1428,7 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
           rvb.startStruct()
           var j = 0
           while (j < keepIndices.length) {
-            rvb.addField(localEntryType, rv.region, eltOffset, j)
+            rvb.addField(localEntryType, rv.region, eltOffset, keepIndices(j))
             j += 1
           }
           rvb.endStruct()

